.github/workflows/build.yml

```diff
@@ -0,0 +1,172 @@
+name: Build
+
+on:
+  push:
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build-web:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        name: Clone
+      - uses: actions/setup-node@v1
+        name: Setup Node
+        with:
+          node-version: "*"
+      - name: Install Yarn
+        run: corepack enable
+      - name: Install npm dependencies
+        run: yarn install
+      - name: Build
+        run: yarn run build
+      - uses: actions/upload-artifact@v3
+        name: Upload Artifacts
+        with:
+          name: tauri-web
+          path: dist/*
+  build-linux:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        name: Clone
+      - uses: actions-rs/toolchain@v1
+        name: Setup Rust
+        with:
+          toolchain: stable
+          override: true
+      - uses: actions/setup-node@v1
+        name: Setup Node
+        with:
+          node-version: "*"
+      - name: Install Yarn
+        run: corepack enable
+      - name: Install npm dependencies
+        run: yarn install
+      - name: Install tauri dependencies
+        run: sudo apt update && sudo apt install -y build-essential libpython3-dev libdbus-1-dev libgtk-3-dev libsoup-gnome2.4-dev libwebkit2gtk-4.0-dev
+      - name: Build Tauri
+        run: yarn run build:tauri
+      - uses: actions/upload-artifact@v3
+        name: Upload Artifacts (appimage)
+        with:
+          name: tauri-linux
+          path: src-tauri/target/release/bundle/appimage/*.AppImage
+      - uses: actions/upload-artifact@v3
+        name: Upload Artifacts (deb)
+        with:
+          name: tauri-debian
+          path: src-tauri/target/release/bundle/deb/*.deb
+  build-windows:
```

Comment:
Can these (`build-linux`, `build-windows`, `build-macos`) all be bundled via an OS matrix? The uploads are different, but I think `upload-artifact` also supports multiple paths (and name can be parametrized on the OS)

Reply:
Yes, i just preferred it this way, cause of the different artifact count and packages needed, but with some conditional tasks, this could be modeled as an os-matrix